pkg/agent/svid/store/service.go

```diff
@@ -0,0 +1,270 @@
+			s.processRecords(ctx)
+		case <-ctx.Done():
+			return nil
+		}
+	}
+}
+
+// deleteSVID deletes a secret using SVIDStore plugin, it gets plugin name from entry selectors
+func (s *SVIDStoreService) deleteSVID(ctx context.Context, log logrus.FieldLogger, entry *common.RegistrationEntry) bool {
+	log = log.WithField(telemetry.Entry, entry.EntryId)
+
+	storeName, err := getStoreName(entry.Selectors)
+	if err != nil {
+		log.WithError(err).Error("selector contains invalid store name")
+		return false
+	}
+
+	log = log.WithField(telemetry.SVIDStore, storeName)
+	svidStore, ok := s.svidStores[storeName]
+	if !ok {
+		log.Error("svid store not found")
+		return false
+	}
+
+	var secretData []string
+	for _, selector := range entry.Selectors {
+		secretData = append(secretData, selector.Value)
+	}
+	if err := svidStore.DeleteX509SVID(ctx, secretData); err != nil {
+		log.WithError(err).Error("failed to delete secret")
+		return false
+	}
+
+	log.Debug("Secret deleted successfully")
+	return true
+}
+
+// storeSVID creates or update a secret using SVIDStore plugin, it gets plugin name from entry selectors
+func (s *SVIDStoreService) storeSVID(ctx context.Context, log logrus.FieldLogger, record *storecache.Record) {
+	if record.Svid == nil {
+		// Svid is not yet provided.
+		return
+	}
+	log = log.WithField(telemetry.Entry, record.Entry.EntryId)
+
+	storeName, err := getStoreName(record.Entry.Selectors)
+	if err != nil {
+		log.WithError(err).Error("selector contains invalid store name")
+		return
+	}
+
+	log = log.WithField(telemetry.SVIDStore, storeName)
+	svidStore, ok := s.svidStores[storeName]
+	if !ok {
+		log.Error("svid store not found")
+		return
+	}
+
+	req, err := s.requestFromRecord(record)
+	if err != nil {
+		log.WithError(err).Error("failed to parse record")
+		return
+	}
+
+	if err := svidStore.PutX509SVID(ctx, req); err != nil {
+		log.WithError(err).Error("failed to Put X509-SVID")
+		return
+	}
+
+	// Set revision, since secret was updated successfully
+	s.cache.HandledRecord(record.Entry, record.Revision)
+	log.Debug("SVID stored successfully")
+}
+
+// TODO: may we change log.Error for debug?
+func (s *SVIDStoreService) processRecords(ctx context.Context) {
+	counter := telemetry_store.StartStoreSVIDUpdates(s.metrics)
+	defer counter.Done(nil)
+
+	for _, record := range s.cache.ReadyToStore() {
+		log := s.log.WithField(telemetry.Revision, record.Revision)
+
+		//  Is entry is marked as delete
+		if record.Entry == nil {
+			// TODO: add a retry backoff
+			if s.deleteSVID(ctx, log, record.HandledEntry) {
+				// Deleted successfully update revision
+				s.cache.HandledRecord(record.HandledEntry, record.Revision)
+			}
+			continue
+		}
+
+		// Entries with changes on selectors must be removed before secret is stored.
+		if record.HandledEntry != nil {
+			// Verify if selector change, and if it changes delete secret from store before update
+			if !reflect.DeepEqual(record.Entry.Selectors, record.HandledEntry.Selectors) {
```

Comment:
reflect.DeepEqual may not be the most performant option. We already have a function to compare selectors that we may use: https://github.com/spiffe/spire/blob/master/pkg/common/util/sort.go#L71

Reply:
Added a new function to compare 2 selectors slice, and using it